tools/satsolver-main.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libsolutil/CDCL.h>
+#include <vector>
+#include <string>
+#include <iostream>
+#include <iomanip>
+#include <fstream>
+#include <sstream>
+
+using namespace solidity::util;
+using std::vector;
+using std::string;
+using std::cout;
+using std::endl;
+using std::optional;
+using std::nullopt;
+using std::make_pair;
+
+const int verbose = 0;
+vector<string> cutStringBySpace(string& input);
+optional<vector<Literal>> parseLine(std::string& line);
+std::pair<vector<vector<Literal>>, size_t> readCNFFile(const string& fname);
+size_t getNumVars(const vector<vector<Literal>>& cls);
+
+
+vector<string> cutStringBySpace(string& input)
+{
+    vector<string> parts;
+
+    size_t pos = 0;
+    while ((pos = input.find(" ")) != string::npos)
+	{
+        parts.push_back(input.substr(0, pos));
+        input.erase(0, pos + 1);
+    }
+    parts.push_back(input);
+    return parts;
```

Comment:
```suggestion
	return input | ranges::views::split(' ') | ranges::to<vector<string>>();
```
(or just use it inline)

Reply:
Hmmm this doesn't seem to compile:
```
/home/matesoos/development/solidity/tools/satsolver-main.cpp:46:66: error: ‘to’ is not a member of ‘std::ranges’
```

And indeed, [it doesn't seem to be](https://en.cppreference.com/w/cpp/ranges)? Maybe it's a typo? I can't seem to be able to find the right function, though. Which function did you mean?

Reply:
Ah, now I see! Thanks to both you and @ekpyron , it's now fixed and MUCH cleaner :) Thanks again! Fix in: 3822881f7bcbe38d6178216c30f848b7ecd8b25a